src/network/addrthread.py

```diff
@@ -15,15 +21,25 @@ def run(self):
             while True:
                 try:
                     data = addrQueue.get(False)
-                    chunk.append((data[0], data[1]))
-                    if len(data) > 2:
-                        source = BMConnectionPool().getConnectionByAddr(data[2])
+                    chunk.append(data)
                 except Queue.Empty:
                     break
-                except KeyError:
-                    continue
 
-            # finish
+            if chunk:
+                # Choose peers randomly
+                connections = BMConnectionPool().establishedConnections()
+                randomshuffle(connections)
+                for i in connections:
+                    randomshuffle(chunk)
+                    filtered = []
+                    for stream, peer, seen, destination in chunk:
+                        if peer in (destination, i.destination):
```

Comment:
I'm not sure. What is it mean if `peer == destination`? The node sent it's own address? But what if `i.destination == destination`?

Reply:
`peer == destination` means peer shouldn't receive its own address.
`peer == i.destination` means a peer shouldn't receive an address back that it just sent to me, apart from it being wasteful in general, if both sides used the same algorithm, it would cause infinite loops

Reply:
or maybe I didn't do it correctly

Reply:
yea the logic is wrong

Reply:
I thought `destination` here is the **source** of `peer`.

Reply:
The labels are confusing in this case. Here, `peer` is a variable from the list, while `i.destination` is the remote IP of the connection you're assembling the message for. But in general you are right, the code from the 29f99e6 you commented on was incorrect, I fixed it in a69732f.